### ISSUE TITLE
add an option for go flags in go/build pipeline

### ISSUE THREAD

**Issue by developer-guy** [chainguard-dev/melange#474]
We (w/@dentrax) thought that if people want to add additional flags for the Go build, we should provide an option for this in the melange go/build pipeline, but there is no option for that yet.

AFAIK, there is an environment variable called `GOFLAGS` on go side that lets you specify the build arguments without editing the command itself but I'm not sure it would work if we set this in build process, if so, please feel free to close this issue.

Final implementation would be something like the following:

```shell
  - uses: go/build
    with:
      modroot: cluster-autoscaler
      packages: .
      output: cluster-autoscaler
      ldflags: -s -w
      flags: -mod=vendor <-- additional option for setting Go build flags.
```

**Comment by debasishbsws** [comment]
Yes, we can specify the [`GOFLAGS`](https://go-review.googlesource.com/c/go/+/126656/4) env var and it will pass it as a flag in every build command. But, I think will not work here as go/build is in a different pipeline and we can not carry forward exported env vars in the previous runs.

And this additional flags could be a list that we ca pass into the command
